### PR TITLE
feat(layer2): add integration Boba Network

### PR DESCRIPTION
**Pull request by developerfred**
### Bridge  Mainet 
- https://gateway.boba.network/
- https://docs.boba.network/user-docs/how-to-bridge

### Dex 
- https://oolongswap.com/#/stake
- https://app.senpaiswap.com/#/swap
- https://swapperchan.com/en/swap

### Maskbook Token List 
- [ ] https://github.com/DimensionDev/assets/pull/3

### Contracts Boba Mainnet 
-  [x] [MysteryBox]()
- [x] [HappyTokenPool]()
- [x] [HappyRedPacket]()
- [x] [BalanceChecker]
- [x] [HappyRedPacket_ERC721 ]

### Contracts Boba Rinkeby

- [MysteryBox (0xf44799–8ff9e9)](https://blockexplorer.rinkeby.boba.network/address/0xf44799d9f0ab280a5f26fd3ab5557595948ff9e9)
- [HappyRedPacket (0xb4d669–705b21)](https://blockexplorer.rinkeby.boba.network/address/0xb4d669bc117735fda44e90e52795132187705b21)
- [HappyRedPacket_ERC721 (0xdca6f4–c209b3)](https://blockexplorer.rinkeby.boba.network/address/0xdca6f476eebcde8fe8b072e3fc80dbc28dc209b3)
- [HappyTokenPool (0xd60a38–a94ad8)](https://blockexplorer.rinkeby.boba.network/address/0xd60a384b01ac271b68483441a06ba32b96a94ad8)
- [BalanceChecker (0x269090–8a7e9c)](https://blockexplorer.rinkeby.boba.network/address/0x2690906609a9af3bf08b8384cd9a0f97a88a7e9c)
- [UniswapInterfaceMulticall (0xdb93cc–c00fdc)](https://blockexplorer.rinkeby.boba.network/address/0xdb93ccd481012bb5d1e2c8d0af7c5f2940c00fdc)

### block Explorer
https://blockexplorer.boba.network/

## docs
https://docs.boba.network/developer-docs/005_parameters-rinkeby

## Boba Interceptor  
https://github.com/DimensionDev/Maskbook/pull/5644

### CODE REVIEW

**Comment by UncleBill** [comment]
Hey, @developerfred , you should set the target of a pull request to `develop` branch.

File: packages/mask/src/plugins/ITO/SNSAdaptor/ClaimAllDialog.tsx [hunk 1]
```diff
@@ -351,6 +358,27 @@ export function ClaimAllDialog(props: ClaimAllDialogProps) {
...
+            createTabItem('ETH', ChainId.Mainnet),
+            createTabItem('BSC', ChainId.BSC),
+            createTabItem('Polygon/Matic', ChainId.Matic),
+            createTabItem('Arbitrum', ChainId.Arbitrum),
+            createTabItem('xDai', ChainId.xDai),
+            createTabItem('Boba', ChainId.Boba),
+        ],
+        index: [ChainId.Mainnet, ChainId.BSC, ChainId.Matic, ChainId.Arbitrum, ChainId.xDai, ChainId.Boba].indexOf(
+            chainId,
+        ),
```
Comment (septs) [inline packages/mask/src/plugins/ITO/SNSAdaptor/ClaimAllDialog.tsx]: plz extract chain id list as table-driven programming

e.g:
```ts
const tabItems = {
   [ChainId.Mainnet]: "ETH",
   // ... eth ...
}

tabs: Object.entries(tabItems).map(([chainId, displayName]) => createTabItem(displayName, chainId)),
index: Object.keys(tabItems).indexOf(chainId)
```

File: packages/mask/src/plugins/ITO/SNSAdaptor/ClaimAllDialog.tsx [hunk 2]
```diff
@@ -227,9 +228,15 @@ export function ClaimAllDialog(props: ClaimAllDialogProps) {
...
-        )
+        [
+            ChainId.Mainnet,
+            ChainId.BSC,
+            ChainId.Matic,
+            ChainId.Arbitrum,
+            ChainId.xDai,
+            ChainId.Boba,
+            ChainId.Fantom,
+        ].includes(currentChainId)
```
Comment (septs) [inline packages/mask/src/plugins/ITO/SNSAdaptor/ClaimAllDialog.tsx]: plz extract the list as constants to top-level

File: packages/mask/src/plugins/Trader/trader/useGetTradeContext.ts [hunk 3]
```diff
@@ -113,6 +113,18 @@ export function useGetTradeContext(tradeProvider?: TradeProvider) {
...
+                    TYPE: tradeProvider,
+                    IS_UNISWAP_V2_LIKE: true,
+                    GRAPH_API: getTraderConstants(chainId).OOLONGSWAP_THEGRAPH,
+                    INIT_CODE_HASH: getTraderConstants(chainId).OOLONGSWAP_INIT_CODE_HASH,
+                    ROUTER_CONTRACT_ADDRESS: getTraderConstants(chainId).OOLONGSWAP_ROUTER_ADDRESS,
+                    FACTORY_CONTRACT_ADDRESS: getTraderConstants(chainId).OOLONGSWAP_FACTORY_ADDRESS,
+                    AGAINST_TOKENS: SUSHISWAP_BASE_AGAINST_TOKENS,
+                    ADDITIONAL_TOKENS: {},
+                    CUSTOM_TOKENS: SUSHISWAP_CUSTOM_BASES,
+                }
```
Comment (septs) [inline packages/mask/src/plugins/Trader/trader/useGetTradeContext.ts]:
```suggestion
            case TradeProvider.OOLONGSWAP:
                const constants = getTraderConstants(chainId)
                return {
                    TYPE: tradeProvider,
                    IS_UNISWAP_V2_LIKE: true,
                    GRAPH_API: constants.OOLONGSWAP_THEGRAPH,
                    INIT_CODE_HASH: constants.OOLONGSWAP_INIT_CODE_HASH,
                    ROUTER_CONTRACT_ADDRESS: constants.OOLONGSWAP_ROUTER_ADDRESS,
                    FACTORY_CONTRACT_ADDRESS: constants.OOLONGSWAP_FACTORY_ADDRESS,
                    AGAINST_TOKENS: SUSHISWAP_BASE_AGAINST_TOKENS,
                    ADDITIONAL_TOKENS: {},
                    CUSTOM_TOKENS: SUSHISWAP_CUSTOM_BASES,
                }
```

File: packages/mask/src/plugins/Trader/trader/useTradeContext.ts [hunk 4]
```diff
@@ -109,6 +109,18 @@ export function useTradeContext(tradeProvider: TradeProvider) {
...
+                    TYPE: tradeProvider,
+                    IS_UNISWAP_V2_LIKE: true,
+                    GRAPH_API: getTraderConstants(chainId).OOLONGSWAP_THEGRAPH,
+                    INIT_CODE_HASH: getTraderConstants(chainId).OOLONGSWAP_INIT_CODE_HASH,
+                    ROUTER_CONTRACT_ADDRESS: getTraderConstants(chainId).OOLONGSWAP_ROUTER_ADDRESS,
+                    FACTORY_CONTRACT_ADDRESS: getTraderConstants(chainId).OOLONGSWAP_FACTORY_ADDRESS,
+                    AGAINST_TOKENS: SUSHISWAP_BASE_AGAINST_TOKENS,
+                    ADDITIONAL_TOKENS: {},
+                    CUSTOM_TOKENS: SUSHISWAP_CUSTOM_BASES,
+                }
```
Comment (septs) [inline packages/mask/src/plugins/Trader/trader/useTradeContext.ts]: https://github.com/DimensionDev/Maskbook/pull/5310/files#r778776159

extract `getTraderConstants(chainId)` as variable

File: packages/maskbook/src/plugins/Trader/trader/useTradeComputed.ts [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+    strategy: TradeStrategy,
+    inputAmount: string,
+    outputAmount: string,
+    inputToken?: FungibleTokenDetailed,
+    outputToken?: FungibleTokenDetailed,
+) {
+    const inputTokenProduct = pow10(inputToken?.decimals ?? 0)
+    const outputTokenProduct = pow10(outputToken?.decimals ?? 0)
+    const inputAmount_ = new BigNumber(inputAmount || '0').multipliedBy(inputTokenProduct).integerValue().toFixed()
+    const outputAmount_ = new BigNumber(outputAmount || '0').multipliedBy(outputTokenProduct).integerValue().toFixed()
```
Comment (septs) [inline packages/maskbook/src/plugins/Trader/trader/useTradeComputed.ts:33]:
```suggestion
    const inputAmount_ = rightShift(inputAmount || '0', inputToken?.decimals ?? 0).integerValue().toFixed()
    const outputAmount_ = rightShift(outputAmount || '0', outputToken?.decimals ?? 0).integerValue().toFixed()
```

Comment (septs) [inline packages/maskbook/src/plugins/Trader/trader/useTradeComputed.ts]:
```suggestion
export function useTradeComputed(
    provider: TradeProvider,
    strategy: TradeStrategy,
    inputAmount: BigNumber.Value = '0',
    outputAmount: BigNumber.Value = '0',
    inputToken?: FungibleTokenDetailed,
    outputToken?: FungibleTokenDetailed,
) {
    inputAmount = rightShift(inputAmount, inputToken?.decimals ?? 0).integerValue().toFixed()
    outputAmount = rightShift(outputAmount, outputToken?.decimals ?? 0).integerValue().toFixed()
```

File: packages/maskbook/src/plugins/Wallet/apis/debank.ts [hunk 6]
```diff
@@ -0,0 +1,38 @@
+import type { BalanceListResponse, HistoryResponse, GasPriceDictResponse } from '../types'
+import urlcat from 'urlcat'
+import { ChainId } from '@masknet/web3-shared-evm'
+
+const DEBANK_API = 'https://api.debank.com'
+
+export async function getTransactionList(address: string, chain: string) {
+    const response = await fetch(`${DEBANK_API}/history/list?user_addr=${address.toLowerCase()}&chain=${chain}`)
+    return (await response.json()) as HistoryResponse
+}
```
Comment (septs) [inline packages/maskbook/src/plugins/Wallet/apis/debank.ts]:
```suggestion
export async function getTransactionList(address: string, chain: string): Promise<HistoryResponse> {
    const response = await fetch(urlcat(DEBANK_API, '/history/list', { user_addr: address.toLowerCase(), chain }))
    return response.json()
}
```

File: packages/maskbook/src/plugins/Wallet/apis/debank.ts [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+export async function getAssetsList(address: string) {
+    const response = await fetch(`${DEBANK_API}/token/balance_list?is_all=true&user_addr=${address.toLowerCase()}`)
+    return (await response.json()) as BalanceListResponse
+}
```
Comment (septs) [inline packages/maskbook/src/plugins/Wallet/apis/debank.ts]:
```suggestion
export async function getAssetsList(address: string): Promise<BalanceListResponse> {
    const response = await fetch(DEBANK_API, '/token/balance_list', { is_all: true, user_addr: address.toLowerCase() })
    return response.json()
}
```

File: packages/maskbook/src/plugins/Wallet/apis/debank.ts [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+
+const chainIdMap: Record<number, string> = {
```
Comment (septs) [inline packages/maskbook/src/plugins/Wallet/apis/debank.ts]:
```suggestion
const chainIdMap: Record<ChainId, string> = {
```

File: packages/maskbook/src/plugins/Wallet/apis/debank.ts [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+    [ChainId.Mainnet]: 'eth',
+    [ChainId.BSC]: 'bsc',
+    [ChainId.xDai]: 'xdai',
+    [ChainId.Matic]: 'matic',
+    [ChainId.Arbitrum]: 'arb',
+    [ChainId.Boba]: 'bob',
+}
+
+const getDebankChain = (chainId: number) => {
```
Comment (septs) [inline packages/maskbook/src/plugins/Wallet/apis/debank.ts]:
```suggestion
const getDebankChain = (chainId: ChainId) => {
```

File: packages/maskbook/src/plugins/Wallet/apis/debank.ts [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+    return chainIdMap[chainId] ?? ''
+}
+
+export async function getGasPriceDict(chainId: ChainId) {
+    const chain = getDebankChain(chainId)
+    const response = await fetch(urlcat(DEBANK_API, '/chain/gas_price_dict_v2', { chain }))
+    const result = await response.json()
+    if (result.error_code === 0) {
+        return result as GasPriceDictResponse
+    }
```
Comment (septs) [inline packages/maskbook/src/plugins/Wallet/apis/debank.ts]:
```suggestion
    const result: GasPriceDictResponse = await response.json()
    if (result.error_code === 0) {
        return result
    }
```

File: packages/maskbook/src/plugins/Wallet/pipes/index.ts [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+            chainId: ChainId.Mainnet,
+        },
+        'bsc-assets': {
+            chain: 'bsc',
+            chainId: ChainId.BSC,
+        },
+        'polygon-assets': {
+            chain: 'matic',
+            chainId: ChainId.Matic,
+        },
```
Comment (septs) [inline packages/maskbook/src/plugins/Wallet/pipes/index.ts]:
```suggestion
        assets: { chain: 'eth', chainId: ChainId.Mainnet },
        'bsc-assets': { chain: 'bsc', chainId: ChainId.BSC },
        'polygon-assets': { chain: 'matic', chainId: ChainId.Matic },
```

File: packages/shared/src/UI/components/NetworkIcon/index.tsx [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+
+export function NetworkIcon(props: NetworkIconProps) {
+    const { size = 48, networkType } = props
+    const classes = useStylesExtends(useStyles(), props)
+
+    if (!networkType) return null
+
+    return icons[networkType] ? (
+        <img height={size} width={size} src={icons[networkType]} className={classes.icon} />
+    ) : null
```
Comment (septs) [inline packages/shared/src/UI/components/NetworkIcon/index.tsx]:
```suggestion
    if (!icons[networkType]) return null
    return <img height={size} width={size} src={icons[networkType]} className={classes.icon} />
```

File: packages/web3-constants/evm/token.json [hunk 13]
```diff
@@ -1,4 +1,21 @@
 {
+    "ZERO_ADDRESS": {
```
Comment (septs) [inline packages/web3-constants/evm/token.json:2]: plz sync develop branch.

https://github.com/DimensionDev/Maskbook/blob/develop/packages/web3-constants/evm/token.json

File: packages/mask/src/plugins/ITO/SNSAdaptor/ClaimAllDialog.tsx [hunk 14]
```diff
@@ -221,11 +221,15 @@ export function ClaimAllDialog(props: ClaimAllDialogProps) {
     const DialogRef = useRef<HTMLDivElement>(null)
     const account = useAccount()
     const currentChainId = useChainId()
-    const {
-        value: campaignInfos,
-        loading: loadingAirdrop,
-        retry: retryAirdrop,
-    } = useSpaceStationCampaignInfo(account, Flags.nft_airdrop_enabled)
+
+    const tabItems: Record<number, string> = {
```
Comment (septs) [inline packages/mask/src/plugins/ITO/SNSAdaptor/ClaimAllDialog.tsx]:
```suggestion
    const tabItems: Record<ChainId, string> = {
```

Comment (developerfred) [reply packages/mask/src/plugins/ITO/SNSAdaptor/ClaimAllDialog.tsx]: done

File: packages/mask/src/plugins/ITO/SNSAdaptor/ClaimAllDialog.tsx [hunk 15]
```diff
@@ -220,19 +221,38 @@ export function ClaimAllDialog(props: ClaimAllDialogProps) {
...
+            ChainId.Matic,
+            ChainId.Arbitrum,
+            ChainId.xDai,
+            ChainId.Boba,
+            ChainId.Fantom,
+            ChainId.Celo,
+        ].includes(currentChainId)
+            ? currentChainId
+            : ChainId.Mainnet,
+    )
```
Comment (septs) [inline packages/mask/src/plugins/ITO/SNSAdaptor/ClaimAllDialog.tsx]:
```suggestion
    const [chainId, setChainId] = useState(tabItems[currentChainId] ? currentChainId : ChainId.Mainnet)
```

File: packages/mask/src/plugins/Trader/trader/0x/useTrade.ts [hunk 16]
```diff
@@ -30,6 +30,8 @@ export function getNativeTokenLabel(networkType: NetworkType) {
             return '0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee'
         case NetworkType.xDai:
             return '0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee'
+        case NetworkType.Boba:
```
Comment (septs) [inline packages/mask/src/plugins/Trader/trader/0x/useTrade.ts]: https://github.com/DimensionDev/Maskbook/pull/5396/files

File: cspell.json [hunk 17]
```diff
@@ -55,10 +55,16 @@
         "Vcent",
         "Viet",
         "aave",
+        "Adai",
```
Comment (septs) [inline cspell.json]: plz resync cspell.json file

see #5420

Comment (developerfred) [reply cspell.json]: done @septs

File: cspell.json [hunk 18]
```diff
@@ -301,6 +304,7 @@
         "inpage",
         "irss",
         "isfacebook",
+        "jkoeaghipilijlahjplgbfiocjhldnap",
```
Comment (septs) [inline cspell.json]: why?

Comment (developerfred) [reply cspell.json]: check-spell (pull_request) fails on file : [evm-integration](https://github.com/DimensionDev/Maskbook/edit/develop/docs/evm-integration.md) 

Goto `chrome-extension://jkoeaghipilijlahjplgbfiocjhldnap/dashboard.html#/wallets/transfer` and check the estimated USD value is working.

Comment (septs) [reply cspell.json]: CC @guanbinrui @Jack-Works

Comment (UncleBill) [reply cspell.json]: I think it's OK to ignore the extension id.
Or we can use inline ignore, like

```md
<!-- cspell:ignore jkoeaghipilijlahjplgbfiocjhldnap -->
```

File: packages/mask/src/plugins/Trader/trader/openocean/useTrade.ts [hunk 19]
```diff
@@ -36,7 +36,7 @@ export function useTrade(
         if (!inputToken || !outputToken) return null
         if (inputAmount === '0') return null
         const sellToken = isNativeTokenAddress(inputToken.address)
-            ? { ...inputToken, address: OPENOCEAN_ETH_ADDRESS ?? '' }
+            ? { ...inputToken, address: OPENOCEAN_ETH_ADDRESS ?? ' ' }
```
Comment (septs) [inline packages/mask/src/plugins/Trader/trader/openocean/useTrade.ts]:
```suggestion
            ? { ...inputToken, address: OPENOCEAN_ETH_ADDRESS ?? '' }
```

File: packages/web3-constants/evm/token-asset-base-url.json [hunk 20]
```diff
@@ -39,6 +39,11 @@
         "Arbitrum": [],
         "Arbitrum_Rinkeby": [],
         "xDai": [],
+        "Boba": [
+            "https://raw.githubusercontent.com/dimensiondev/assets/master/blockchains/boba",
```
Comment (UncleBill) [inline packages/web3-constants/evm/token-asset-base-url.json:43]: We don't have assets for `Boba` in the assets repo yet, please help to collect them and send a PR there.

https://github.com/DimensionDev/assets/tree/master/blockchains

**Review by UncleBill — COMMENTED** [review]
I am testing this PR.

File: packages/web3-shared/evm/assets/chains.json [hunk 21]
```diff
@@ -335,6 +335,56 @@
         "chainId": 10,
         "networkId": 10
     },
+    {
+        "name": "Boba Network",
+        "chain": "BOBA",
```
Comment (UncleBill) [inline packages/web3-shared/evm/assets/chains.json]:
```suggestion
        "chain": "Boba",
```


Please make sure this button will not show "Wrong network"
image

File: packages/web3-shared/evm/assets/chains.json [hunk 22]
```diff
@@ -335,6 +335,56 @@
...
+            {
+                "name": "blockexplorer",
+                "url": "https://blockexplorer.boba.network/",
+                "standard": "EIP3091"
+            }
+        ]
+    },
+    {
+        "name": "Boba Network Rinkeby",
+        "chain": "BOBA",
```
Comment (UncleBill) [inline packages/web3-shared/evm/assets/chains.json]:
```suggestion
        "chain": "Boba",
```

**Review by UncleBill — COMMENTED** [review]
### UniswapInterfaceMulticall

update multicall contract address by following tabl.

| Chain            | Address                                                                                  |
| ---------------- | ---------------------------------------------------------------------------------------- |
| Mainnet          | [0x1F98415757620B543A52E61c46B32eB19261F984][UniswapInterfaceMulticall-mainnet]          |
| Ropsten          | [0x1F98415757620B543A52E61c46B32eB19261F984][UniswapInterfaceMulticall-ropsten]          |
| Rinkeby          | [0x1F98415757620B543A52E61c46B32eB19261F984][UniswapInterfaceMulticall-rinkeby]          |
| BSC              | [0x5211fe40fE49a13CC86E3bC77855152457eDE585][UniswapInterfaceMulticall-bsc]              |
| Matic            | [0x54c06a5D7C6447BE47E07390eD7da7e6994585f4][UniswapInterfaceMulticall-matic]            |
| Arbitrum         | [0xadF885960B47eA2CD9B55E6DAc6B42b7Cb2806dB][UniswapInterfaceMulticall-arbitrum]         |
| Rinkeby-Arbitrum | [0xa501c031958F579dB7676fF1CE78AD305794d579][UniswapInterfaceMulticall-rinkeby-explorer] |
| xDai             | [0xAb7B1bE4233A04e5C43a810E75657ECED8E5463B][UniswapInterfaceMulticall-xdai]             |
| Fantom           | [0x913975af2Bb8a6Be4100D7dc5e9765B77F6A5d6c][UniswapInterfaceMulticall-fantom]           |
| Celo             | [0x6cc1b1058F9153358278C35E0b2D382f1585854B][UniswapInterfaceMulticall-celo]             |
| avalanche        | [0x6cc1b1058F9153358278C35E0b2D382f1585854B][UniswapInterfaceMulticall-avalanche]        |
| Aurora-testnet   | [0x7e0Ac0A9341017a687f8d8bD4287104Ea86c5924][UniswapInterfaceMulticall-aurora-test]      |
| Aurora           | [0x6cc1b1058F9153358278C35E0b2D382f1585854B][UniswapInterfaceMulticall-aurora]           |
| Fuse             | [0x6cc1b1058F9153358278C35E0b2D382f1585854B][UniswapInterfaceMulticall-fuse]             |
| Metis            | [0x6cc1b1058F9153358278C35E0b2D382f1585854B][UniswapInterfaceMulticall-metis]            |
| Optimistic       | [0x561c5f3a19871ecb1273D6D8eCc276BeEDa5c8b4][UniswapInterfaceMulticall-Optimistic]       |
| Boba             | [0x6cc1b1058F9153358278C35E0b2D382f1585854B][UniswapInterfaceMulticall-Boba]             |
| Moonriver        | [0x6cc1b1058F9153358278C35E0b2D382f1585854B][UniswapInterfaceMulticall-Moonriver]   |

[UniswapInterfaceMulticall-mainnet]: https://etherscan.io/address/0x1F98415757620B543A52E61c46B32eB19261F984
[UniswapInterfaceMulticall-ropsten]: https://ropsten.etherscan.io/address/0x1F98415757620B543A52E61c46B32eB19261F984
[UniswapInterfaceMulticall-rinkeby]: https://rinkeby.etherscan.io/address/0x1F98415757620B543A52E61c46B32eB19261F984
[UniswapInterfaceMulticall-bsc]: https://bscscan.com/address/0x5211fe40fE49a13CC86E3bC77855152457eDE585
[0xxxxxxxxxxxxxxxxxxxxxxxxxxxxx]: https://testnet.bscscan.com/address/0xxxxxxxxxxxxxxxxxxxxxxxxxxxxx
[UniswapInterfaceMulticall-matic]: https://explorer-mainnet.maticvigil.com/address/0x54c06a5D7C6447BE47E07390eD7da7e6994585f4
[0xxxxxxxxxxxxxxxxxxxxxxxxxxxxx]: https://explorer-mumbai.maticvigil.com/address/0xxxxxxxxxxxxxxxxxxxxxxxxxxxxx
[UniswapInterfaceMulticall-arbitrum]: https://explorer.arbitrum.io/address/0xadF885960B47eA2CD9B55E6DAc6B42b7Cb2806dB
[UniswapInterfaceMulticall-rinkeby-explorer]: https://rinkeby-explorer.arbitrum.io/address/0xa501c031958F579dB7676fF1CE78AD305794d579
[0xxxxxxxxxxxxxxxxxxxxxxxxxxxxx]: https://kovan-optimistic.etherscan.io/address/0xxxxxxxxxxxxxxxxxxxxxxxxxxxxx
[0xxxxxxxxxxxxxxxxxxxxxxxxxxxxx]: https://optimistic.etherscan.io/address/0xxxxxxxxxxxxxxxxxxxxxxxxxxxxx
[UniswapInterfaceMulticall-xdai]: https://blockscout.com/xdai/mainnet/address/0xAb7B1bE4233A04e5C43a810E75657ECED8E5463B/transactions
[UniswapInterfaceMulticall-fantom]: https://ftmscan.com/address/0x913975af2Bb8a6Be4100D7dc5e9765B77F6A5d6c
[UniswapInterfaceMulticall-celo]: https://explorer.celo.org/address/0x6cc1b1058F9153358278C35E0b2D382f1585854B/contracts
[UniswapInterfaceMulticall-avalanche]: https://snowtrace.io/address/0x6cc1b1058F9153358278C35E0b2D382f1585854B
[UniswapInterfaceMulticall-aurora-test]: https://explorer.testnet.aurora.dev/address/0x7e0Ac0A9341017a687f8d8bD4287104Ea86c5924/transactions
[UniswapInterfaceMulticall-aurora]: https://explorer.mainnet.aurora.dev/address/0x6cc1b1058F9153358278C35E0b2D382f1585854B/transactions
[UniswapInterfaceMulticall-fuse]: https://explorer.fuse.io/address/0x6cc1b1058F9153358278C35E0b2D382f1585854B/transactions
[UniswapInterfaceMulticall-metis]: https://andromeda-explorer.metis.io/address/0x6cc1b1058F9153358278C35E0b2D382f1585854B/transactions
[UniswapInterfaceMulticall-Optimistic]: https://optimistic.etherscan.io/address/0x561c5f3a19871ecb1273D6D8eCc276BeEDa5c8b4
[UniswapInterfaceMulticall-Boba]: https://blockexplorer.boba.network/address/0x6cc1b1058F9153358278C35E0b2D382f1585854B/transactions
[UniswapInterfaceMulticall-Moonriver]: https://moonriver.moonscan.io/address/0x6cc1b1058F9153358278C35E0b2D382f1585854B

File: packages/web3-constants/evm/token-asset-base-url.json (hunk 20 above)
Comment (developerfred) [reply packages/web3-constants/evm/token-asset-base-url.json:43]: Sure!

Comment (developerfred) [reply packages/web3-constants/evm/token-asset-base-url.json:43]: @UncleBill  https://github.com/DimensionDev/assets/pull/3

**Comment by UncleBill** [comment]
@DimensionDev/qa Please help to verify.

**Review by Jack-Works — DISMISSED** [review]

**Comment by wenluomask** [comment]
Thanks for your effort. Here are some issues I found:
1. BOBA token is not in our token list yet, please add the contract address and icon;
Screen Shot 2022-02-09 at 12 04 49 PM

2. Token balance didn't show in token list;

3. Failed to swap;
Screen Shot 2022-02-09 at 12 44 05 PM
 
4. Dashboard's asset isn't ready yet;
Screen Shot 2022-02-09 at 2 03 15 PM

**Comment by developerfred** [comment]
@wenluomask  @UncleBill 

**1**-> https://github.com/DimensionDev/Mask-Token-List/pull/59 ✅ 
**2** -> Fixed [1ac5511]
(https://github.com/DimensionDev/Maskbook/pull/5310/commits/1ac55112f46c2686ca591c57cc5e1c01b5a49e42) ✅ 
Screenshot 2022-02-10 at 05 50 33

**3** -> Fixed ✅ 
**4** -> Fixed https://github.com/DimensionDev/Maskbook/pull/5310#issuecomment-1034466415 ✅ 
Screenshot 2022-02-09 at 14 42 05
Screenshot 2022-02-09 at 16 34 36

**Review by Jack-Works — DISMISSED** [review]

**Comment by UncleBill** [comment]
hi, @wenluomask, after updating our proxy server, the assets are correct now.

image

**Comment by wenluomask** [comment]
#1 and #2 still existed.
#3 and #4 fixed.

5. The boba icon on wallet pages will disappear sometimes;
https://user-images.githubusercontent.com/85149605/153341222-3b73f872-be6f-48de-b6ec-d94c1b334cfb.mov

6. Transfer BOBA in Dashboard didn't get the gas limit;

https://user-images.githubusercontent.com/85149605/153341306-b72cbcab-3bdc-4c5e-8c95-f4af4e5c55a0.mov

Screen Shot 2022-02-10 at 1 06 30 PM

**Comment by developerfred** [comment]
@wenluomask can you test the transfer again? bug number 6

**Comment by developerfred** [comment]
```eth_sendTransaction``` and ```eth_getAccounts``` Unsupported JSON-RPC methods on BobaNetwork  ref: https://docs.boba.network/developer-docs/json-rpc-api#unsupported-json-rpc-methods
 
 fixed here [7e2e385](https://github.com/DimensionDev/Maskbook/pull/5310/commits/7e2e385cb375936e8f9c7e2232fb6a6373c7c846)

**Comment by developerfred** [comment]
@UncleBill @Jack-Works @septs Hi guys, I made this change to the ```eth_sendTransaction``` method, can you review it for me?

**Comment by wenluomask** [comment]
Only #1 and #6 were not fixed yet.

In addition, I found out duplicate Eth token in token list.
Screen Shot 2022-02-11 at 5 55 33 PM

**Review by Jack-Works — DISMISSED** [review]

**Review by Jack-Works — DISMISSED** [review]

**Comment by guanbinrui** [comment]
I'm closing this PR. Cause it doesn't belong to any bounty issue.

**Comment by developerfred** [comment]
@guanbinrui ok if one day I enter the reward program do I need to reopen?

**Comment by guanbinrui** [comment]
> @guanbinrui ok if one day I enter the reward program do I need to reopen?

Yes, please link it if any bounty issue opened.

**Pull request closed without merging**